Widgets/openGL_widgets/CubicGLContext.py

```diff
@@ -6,80 +6,33 @@
 from PyQt5.Qt import Qt
 
 from cython_modules.cython_parse import getLayerOutline, genCubes
-from cython_modules.color_policy import multi_iteration_normalize
 
 from Widgets.openGL_widgets.AbstractGLContext import AbstractGLContext
+from pattern_types.Patterns import AbstractGLContextDecorators
 
 from ColorPolicy import ColorPolicy
 from multiprocessing import Pool
 
-from PIL import Image
-
 class CubicGLContext(AbstractGLContext, QWidget):
     def __init__(self, data_dict):
         super().__init__()
-        self.drawing_function = None
-        self.steps = 1
-        self.spacer = 0.2
-        self.vectors_list = None
+        super().shareData(**data_dict)
         self.vertices = 0
-
         self.buffers = None
         self.buffer_len = 0
-        self.shareData(**data_dict)
```

Comment:
Same as before, don't focus on refactoring unless necessary, for now we have more important before release issues.

Reply:
I dont, it is just that these changes make code clearer since in newest pull requests it becomes bloated even though it is redundantly used across various classes. Code reuse is better since it helps avoiding bugs in the future and it does not much time to fix